variable/comparison.cpp

```diff
@@ -4,16 +4,35 @@
 /// @author Piotr Rozyczko
 #include "scipp/core/element/comparison.h"
 #include "scipp/variable/comparison.h"
+#include "scipp/variable/math.h"
 #include "scipp/variable/transform.h"
+#include "scipp/variable/util.h"
 #include "scipp/variable/variable.h"
 
 using namespace scipp::core;
 
 namespace scipp::variable {
 
-Variable is_approx(const VariableConstView &a, const VariableConstView &b,
-                   const VariableConstView &t) {
-  return transform(a, b, t, element::is_approx);
+Variable is_close(const VariableConstView &a, const VariableConstView &b,
+                  const VariableConstView rtol, const VariableConstView atol,
+                  const NanComparisons equal_nans) {
+  const auto tol = atol + rtol * abs(b);
+  if (a.hasVariances() && b.hasVariances()) {
+    const auto error_tol = atol + rtol * abs(variances(b));
+    if (equal_nans == NanComparisons::Equal)
+      return variable::transform(a, b, tol, element::is_close_equal_nan) &
+             variable::transform(sqrt(variances(a)), sqrt(variances(b)),
+                                 error_tol, element::is_close_equal_nan);
+    else
+      return variable::transform(a, b, tol, element::is_close) &
+             variable::transform(sqrt(variances(a)), sqrt(variances(b)),
+                                 error_tol, element::is_close);
+  } else {
+    if (equal_nans == NanComparisons::Equal)
+      return variable::transform(a, b, tol, element::is_close_equal_nan);
+    else
+      return variable::transform(a, b, tol, element::is_close);
+  }
```

Comment:
I like this solution!
One small thing: Maybe disallow variances in `atol` and `rtol`?

Also, `tol` can have variances now. Is this intentional? I think this is not a problem because `is_close` will just ignore them. But maybe put it into the `if-else` and use `tol = atol + rtol * abs(values(b))` in the first branch.

Reply:
> One small thing: Maybe disallow variances in atol and rtol?

I can explicitly disallow. Maybe that's more expressive. I dont think there is a bug there at present - but the overloaded ValueAndVariance `abs` and `<=` will ignore variances as is.